pyproject.toml

```diff
@@ -278,7 +284,14 @@ ignore = [
     "E731",
     # Sometimes we need List and Tuple
     "UP006",
+    # optimzer are using the 'compile' method which is a built-in
+    "A003",
+    # commented-out code should be allowed
+    "ERA001",
+    # allow pickle
+    "S301",
```

Comment:
what are the purpose of these changes? I think the ruff has been working correctly so far but just curious

Reply:
ERA001 because Ruff complains about commented out print() statements. I often find these useful for debugging

ensemble.py load_folder/save_folder use pickle for serialization. Ruff flags pickle as unsafe for load because it could load arbitrary objects. I suppose we could switch to https://jsonpickle.github.io/. If jsonpickle is acceptable, I can make the change and remove the S301.

Reply:
@arnavsinghvi11 - what's your opinion on jsonpickle? see my comment above...